test/integration/cli-nfs-shared-volumes-create-failure.test.js

```diff
@@ -0,0 +1,149 @@
+/* * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+/*
+ * Copyright (c) 2017, Joyent, Inc.
+ */
+
+//
+// Summary:
+//
+// These tests ensure that when provisioning fails for a volapi storage VM, we
+// get an error message from sdc-docker rather than a message telling us the
+// creation was successful. We "break" provisioning for the 10g package by
+// setting at trait which no CNs will have. The provision will then fail at
+// the workflow job when we're trying to allocate a CN.
+//
+
+var assert = require('assert-plus');
+var common = require('../lib/common');
+var mod_testVolumes = require('../lib/volumes');
+
+assert.string(process.env.DOCKER_CLI_VERSION, 'process.env.DOCKER_CLI_VERSION');
+
+var dockerVersion = common.parseDockerVersion(process.env.DOCKER_CLI_VERSION);
+if (dockerVersion.major < 1 || dockerVersion.minor < 9) {
+    console.log('Skipping volume tests: volumes are not supported in Docker '
+        + 'versions < 1.9');
+    process.exit(0);
+}
+
+if (!mod_testVolumes.nfsSharedVolumesSupported()) {
+    console.log('Skipping volume tests: volumes are not supported in this '
+        + 'Triton setup');
+    process.exit(0);
+}
+
+var test = require('tape');
+
+var cli = require('../lib/cli');
+var h = require('./helpers');
+var volumesCli = require('../lib/volumes-cli');
+
+var createTestVolume = mod_testVolumes.createTestVolume;
+
+var NFS_SHARED_VOLUME_NAMES_PREFIX =
+    mod_testVolumes.getNfsSharedVolumesNamePrefix();
+
+var ALICE_USER;
+var PAPI;
+var PAPI_PACKAGE;
+var PAPI_ORIGINAL_TRAITS;
+
+test('setup', function (tt) {
+    tt.test('DockerEnv: alice init', function (t) {
+        cli.init(t, function onCliInit(err, env) {
+            t.ifErr(err, 'Docker environment initialization should not err');
+            if (env) {
+                ALICE_USER = env.user;
+            }
+        });
+    });
+
+    tt.test('setup PAPI client', function (t) {
+        h.createPapiClient(function (err, _papi) {
+            t.ifErr(err, 'create PAPI client');
+            PAPI = _papi;
+            t.end();
+        });
+    });
+
+    tt.test('getting 10g PAPI package', function (t) {
+        PAPI.list('(&(name=sdc_volume_nfs_10)(active=true))',
+            {},
+            function _onResults(err, pkgs, count) {
+                var results = [];
+                t.ifErr(err, 'get PAPI package');
+                if (pkgs) {
+                    results = pkgs.map(function mapUuids(pkg) {
+                        return pkg.uuid;
+                    });
+                }
+                t.equal(count, 1, 'should be 1 result ' + JSON.stringify(results));
+                if (count === 1 && results.length === 1) {
+                    PAPI_PACKAGE = results[0];
```

Comment:
The reason for doing this was so that when the test failed, we only had the uuids in the assert message, so we can go look at which packages we got that we didn't expect. I'll change to just do the map there I guess.